extensions/amp-viewer-integration/0.1/messaging/messaging.js

```diff
@@ -171,13 +171,14 @@ export class Messaging {
           return;
         }
         if (
-          event.origin == origin &&
+          (event.origin == origin ||
+            event.origin.endsWith('.cdn.ampproject.org')) &&
```

Comment:
Actually, let's change this to check [`cdnProxyRegex`](https://github.com/ampproject/amphtml/blob/e314f8cdc92a0898cbab7ee2483a661cad16cfa9/src/config.js#L74):

```suggestion
            urls.cdnProxyRegex.test(event.origin)) &&
```